crates/ursa-pod/src/server.rs

```diff
@@ -180,26 +176,21 @@ impl<S: AsyncWrite + AsyncRead + Unpin, B: Backend> UfdpHandler<S, B> {
                 self.session_id
             );
 
-            // wait for delivery acknowledgment
+            // Wait for delivery acknowledgment
             match instrument!(
-                self.conn.read_frame(None).await?,
+                self.conn.read_frame(Some(DECRYPTION_KEY_REQ_TAG)).await?,
                 "sid={},tag=read_da,content_size={content_size},block_size={block_size}",
                 self.session_id
             ) {
                 Some(UrsaFrame::DecryptionKeyRequest { .. }) => {
                     // todo: transaction manager (batch and store tx)
                 }
-                Some(f) => {
-                    error!("Terminating asdf, unexpected frame: {f:?}");
-                    self.conn
-                        .termination_signal(Some(Reason::UnexpectedFrame))
-                        .await?;
-                    return Err(UrsaCodecError::UnexpectedFrame(f.tag().unwrap()));
-                }
                 None => return Err(UrsaCodecError::Unknown),
+                Some(_) => unreachable!(),
```

Comment:
Don't trust the client to oblige to the protocol. we should never panic here.

Reply:
the frame filter should guarantee unreachable there